app/src/main/kotlin/com/wire/android/util/permission/CallingRecordAudioRequestFlow.kt

```diff
@@ -37,17 +41,25 @@ class CallingAudioRequestFlow(
     private val audioRecordPermissionLauncher: ManagedActivityResultLauncher<Array<String>, Map<String, @JvmSuppressWildcards Boolean>>
 ) {
     fun launch() {
-        if (context.checkPermission(android.Manifest.permission.RECORD_AUDIO) &&
-            context.checkPermission(android.Manifest.permission.BLUETOOTH_CONNECT)
-        ) {
+        val audioPermissionEnabled = context.checkPermission(android.Manifest.permission.RECORD_AUDIO)
+        val bluetoothPermissionEnabled = context.checkPermission(android.Manifest.permission.BLUETOOTH_CONNECT)
+
+        val neededPermissions = arrayOf(
+            android.Manifest.permission.RECORD_AUDIO
+        )
+
+        val permissionsEnabled = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            neededPermissions[neededPermissions.size] = android.Manifest.permission.BLUETOOTH_CONNECT
+
+            audioPermissionEnabled && bluetoothPermissionEnabled
+        } else {
+            audioPermissionEnabled
+        }
+
+        if (permissionsEnabled) {
             permissionGranted()
         } else {
-            audioRecordPermissionLauncher.launch(
-                arrayOf(
-                    android.Manifest.permission.RECORD_AUDIO,
-                    android.Manifest.permission.BLUETOOTH_CONNECT
-                )
-            )
+            audioRecordPermissionLauncher.launch(neededPermissions)
```

Comment:
This was also changed due to Android versions needing and not needing the bluetooth version, and the way it was before we were always requesting the bluetooth permission even for lower api leves (which no prompt were shown) and thus resulting in always having a false permission for bluetooth for lower levels.